src/parser/jobs/mnk/modules/RiddleOfFire.tsx

```diff
@@ -1,198 +1,208 @@
+			.reduce((previousValue, currentValue) => previousValue.concat(currentValue))
+			.filter(value => value.action === this.celestialRevolution)
 			.length
-	}
 
-	get chakras() {
-		return this.casts.filter(event => [this.data.actions.THE_FORBIDDEN_CHAKRA.id, this.data.actions.ENLIGHTENMENT.id].includes(event.action)).length
+		if (celestialRevolutionUsages > 0) {
+			return new Suggestion({
+				icon: this.celestialRevolution.icon,
+				content: <Trans id="mnk.rof.suggestions.celestialrevolution.content">Avoid using <ActionLink
+					action="CELESTIAL_REVOLUTION"/>.</Trans>,
+				severity: SEVERITY.MAJOR,
+				why: <Trans id="mnk.rof.suggestions.celestialrevolution.why"><ActionLink
+					action="RISING_PHOENIX"/> and <ActionLink action="ELIXIR_FIELD"/> do more potency along with
+					AoE.</Trans>,
+			})
+		}
 	}
 
-	get meditations() {
-		return this.casts.filter(event => event.action === this.data.actions.MEDITATION.id).length
+	output(windows: Array<HistoryEntry<EvaluatedAction[]>>): EvaluationOutput {
+		return {
+			format: 'table',
+			header: {
+				header: <ActionLink showName={false} action={'MASTERFUL_BLITZ'}/>,
+				accessor: 'masterfulblitz',
+			},
+			rows: windows.map((window, i) => {
+				return {
+					actual: this.countBlitzSkills(window),
+					expected: this.numberOfBlitzExpectedInWindow(i, windows),
+				}
+			}),
+		}
 	}
 
-	get stars() {
-		return this.casts.filter(event => event.action === this.data.actions.SIX_SIDED_STAR.id).length
+	private countBlitzSkills(window: HistoryEntry<EvaluatedAction[]>): number {
+		return window.data.filter(value => this.blitzSkills.includes(value.action)).length
 	}
-}
 
-export class RiddleOfFire extends Analyser {
-	static override handle = 'riddleoffire'
-	static override title = t('mnk.rof.title')`Riddle of Fire`
-	static override displayOrder = DISPLAY_ORDER.RIDDLE_OF_FIRE
-
-	@dependency private data!: Data
-	@dependency private suggestions!: Suggestions
-	@dependency private timeline!: Timeline
-
-	private history: Riddle[] = []
-	private riddle?: Riddle
-	private riddleHook?: EventHook<Events['action']>
+	private numberOfBlitzExpectedInWindow(windowIndex: number, windows: Array<HistoryEntry<EvaluatedAction[]>>): number {
+		// First window should always contain 2 blitz
+		if (windowIndex === 0) {
+			return 2
+		}
 
-	private blitzActions: Array<Action['id']> = []
+		const blitzSkillsInLastWindow = this.countBlitzSkills(windows[windowIndex - 1])
+		if (blitzSkillsInLastWindow === 2) {
+			return 1
+		}
+		return 2
+	}
+}
 
-	override initialise(): void {
-		this.blitzActions = fillActions(BLITZ_SKILLS, this.data)
-		const playerFilter = filter<Event>().source(this.parser.actor.id)
-		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.RIDDLE_OF_FIRE.id), this.onGain)
-		this.addEventHook(playerFilter.type('statusRemove').status(this.data.statuses.RIDDLE_OF_FIRE.id), this.onDrop)
+class BrotherhoodEvaluator implements WindowEvaluator {
```

Comment:
At a high level, I'm not sure the value of this. It's not really a matter of if they used BH or not, but if it was wrapped into the window properly without drift. I was thinking more along the lines of a generic DriftEvaluator, so we could track the difference in timing and make sure the full duration of BH was inside RoF. Might be worth talking to #fb-monk people for feedback on this.

Reply:
`DriftEvaluator` doesn't seem to exist? Unless it's something that would need to be created?

Right now since I'm tracking the action of doing BH inside the RoF window I guarantee the full during inside RoF.

I've noticed on my logs that I've been analyzing, sometimes I press BH before RoF because of timing difference, which shows up that I missed BH. I'm guessing the `DriftEvaluator` is what's going to handle that?